[core] Update very old doc message (#53557)
ray-project/ray: doc/source/ray-core/walkthrough.rst

.. note::

>   If ``ray.init()`` isn't called explicitly, it will implicitly be called with no arguments on the first use of a Ray remote API.

Reviewer: Suggested edit for simplicity (removes passive voice and future tense as well)

```suggestion
  Unless you explicitly call ``ray.init()``, the first use of a Ray remote API calls ``ray.init()`` with no arguments.
```

Author: sounds good, just kept the word "implicitly" in there for clarity
